File: src/main/java/cz/stasimek/fakturaceeasypeasy/controller/LoginController.java
package cz.stasimek.fakturaceeasypeasy.controller;

import java.util.HashMap;
import java.util.Map;
import javax.servlet.http.HttpServletRequest;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.security.oauth2.core.user.OAuth2User;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class LoginController {

	@GetMapping("/user")
	public Map<String, Object> user(@AuthenticationPrincipal OAuth2User principal) {
		Map<String, Object> map = new HashMap<>();
		if (principal != null) {
			if (principal.getAttribute("name") != null) {
				map.put("name", principal.getAttribute("name"));
			} else if (principal.getAttribute("login") != null) {
				map.put("name", principal.getAttribute("login"));
			} else if (principal.getAttribute("email") != null) {
				map.put("name", principal.getAttribute("email"));
			}
			//map.put("principal", principal);
		}
		return map;
	}

	@GetMapping("/error")
	public String error(HttpServletRequest request) {
		String message = (String) request.getSession().getAttribute("login.errorMessage");
		request.getSession().removeAttribute("login.errorMessage");
		return message;
	}

}

File: src/main/kotlin/cz/stasimek/fakturaceeasypeasy/controller/LoginController.kt
package cz.stasimek.fakturaceeasypeasy.controller

import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.security.oauth2.core.user.OAuth2User
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.servlet.http.HttpServletRequest

@RestController
@RequestMapping("/api")
class LoginController {

    @GetMapping("/login/user")
    fun user(
        @AuthenticationPrincipal principal: OAuth2User?
    ): Map<String, Any> {
        val map: MutableMap<String, Any> = HashMap()
        if (principal != null) {
            val name = principal.getAttribute<Any?>("name")
            val login = principal.getAttribute<Any?>("login")
            val email = principal.getAttribute<Any?>("email")
            if (name != null) {
                map["name"] = name
            } else if (login != null) {
                map["name"] = login
            } else if (email != null) {
                map["name"] = email
            }
            //map["principal"] = principal;
        }
        return map
    }

    @GetMapping("/login/error")
    fun error(request: HttpServletRequest): String? {
        val message = request.session.getAttribute("login.errorMessage")
        request.session.removeAttribute("login.errorMessage")
        return message as String
    }
}